src/writeFile.js

```diff
@@ -65,13 +65,18 @@ export default function writeFile(globalRef, pattern, file) {
             }
 
             if (!copyUnmodified &&
-                written[file.absoluteFrom] && written[file.absoluteFrom][hash]) {
+                written[file.absoluteFrom] &&
+                written[file.absoluteFrom][file.webpackTo] &&
+                written[file.absoluteFrom][file.webpackTo][hash]
+            ) {
                 info(`skipping '${file.webpackTo}', because it hasn't changed`);
                 return;
             } else {
                 debug(`added ${hash} to written tracking for '${file.absoluteFrom}'`);
                 written[file.absoluteFrom] = {
-                    [hash]: true
+                    [file.webpackTo]: {
+                        [hash]: true
+                    }
                 };
```

Comment:
Do you have a specific use case that can't be accomplished using `copyUnmodified`.

Based on the test case below, you could do the same thing without adding new functionality.

Reply:
I improved the test to add a transform method to the plugin. This way it does not work even if you add `copyUnmodified` option.

For the record, when I had the bug, it was by copying the same file with the same transformation (basically babel + uglify) in two different locations for history purpose (two different files became only one, but both are referenced)